src/main/java/uk/gov/companieshouse/officerfiling/api/validation/AddressValidator.java

```diff
@@ -0,0 +1,171 @@
+package uk.gov.companieshouse.officerfiling.api.validation;
+
+import uk.gov.companieshouse.api.error.ApiError;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.officerfiling.api.enumerations.ApiEnumerations;
+import uk.gov.companieshouse.officerfiling.api.exception.NotImplementedException;
+import uk.gov.companieshouse.officerfiling.api.model.dto.AddressDto;
+import uk.gov.companieshouse.officerfiling.api.model.dto.OfficerFilingDto;
+import uk.gov.companieshouse.officerfiling.api.service.CompanyProfileService;
+import uk.gov.companieshouse.officerfiling.api.validation.error.AddressErrorProvider;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A singular validator to perform validation rules on residential (home) and correspondence (service) addresses.
+ * The validator methods are unaware of which type of address is being validated (ie premises validation has the same rules regardless of the type of address).
+ */
+public class AddressValidator extends OfficerValidator {
+
+    private static final Integer LENGTH_20 = 20;
+    private static final Integer LENGTH_50 = 50;
+    private static final Integer LENGTH_200 = 200;
+    private final List<String> countryList;
+    private final List<String> ukCountryList;
+
+    protected AddressValidator(Logger logger, CompanyProfileService companyProfileService, String inputAllowedNationalities, ApiEnumerations apiEnumerations, List<String> countryList, List<String> ukCountryList) {
+        super(logger, companyProfileService, inputAllowedNationalities, apiEnumerations);
+        this.countryList = countryList;
+        this.ukCountryList = ukCountryList;
+    }
+
+    @Override
+    public void validateRequiredDtoFields(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto) {
```

Comment:
just mark method abstract?

Reply:
But then I would have to make the class abstract? I thought that would be confusing as we're not expecting anything to extend this class

Reply:
It should not extend OfficerValidator. Then you don't need to Override.

Reply:
But then you need access to quite a few methods in OfficerValidator. Maybe not throwing an exception is the better approach if there are no fields to validate. We could validate the two addresses via these methods but I don't think that's right either. I'd prefer not to have the NotImplemented exceptions over having empty methods.

Reply:
But OfficerValidator currently contains a lot of validation methods that are used in the address validation. I'd rather include these two overridden methods than duplicate all those methods in this class. Those methods could be moved somewhere else to be re-used across all the validators but that would require refactoring quite a lot of code in the other validators.

Reply:
To be honest I'd prefer to just remove them from the abstract class as they're not even all implemented in the other validators.

Reply:
That sounds the best plan thanks.